File: exercism/go/hamming/hamming.go
package hamming

import "fmt"

// Distance calculates the hamming distance for two DNA strands
func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return 0, fmt.Errorf("lengths of the two strands are different")
	}

	difference := 0
	for i := 0; i < len(a); i++ {
		if a[i] != b[i] {
			difference++
		}
	}

	return difference, nil
}

File: exercism/c/hamming/src/hamming.c
#include "hamming.h"
#include <string.h>


int compute(const char *lhs, const char *rhs) {
  int difference = 0;

  if (lhs == NULL || rhs == NULL)
    return -1;

  if (strlen(lhs) != strlen(rhs))
    return -1;

  while (*lhs) {
    if (*lhs++ != *rhs++)
      difference++;
  }

  return difference;
}
